src/datadreamer/image_generation/sdxl_turbo_image_generator.py

```diff
@@ -32,13 +32,22 @@ def _init_gen_model(self):
         Returns:
             AutoPipelineForText2Image: The initialized Stable Diffusion Turbo model.
         """
-        base = AutoPipelineForText2Image.from_pretrained(
-            "stabilityai/sdxl-turbo",
-            torch_dtype=torch.float16,
-            variant="fp16",
-            use_safetensors=True,
-        )
-        base.enable_model_cpu_offload()
+        if self.device == "cpu":
+            base = AutoPipelineForText2Image.from_pretrained(
+                "stabilityai/sdxl-turbo",
+                # variant="fp16",
```

Comment:
The same question about `variant` as above.

Reply:
Because it's not supported for the cpu. Right @HonzaCuhel ?

Reply:
Yeah, fp16 is supported only on GPUs, not CPUs.